### PR TITLE
Reduce docker image size and fix libpng CVEs

### PR DESCRIPTION
**Pull request by adamgibbins**
Upgrades python to pull in newer alpine which fixes a load of  libpng CVEs (https://snyk.io/vuln/SNYK-LINUX-LIBPNG-174098 and https://snyk.io/vuln/SNYK-LINUX-LIBPNG-174252).

Additionally removes apk cache to remove need to clean it up later and purges pyc files to reduce the size.

### CODE REVIEW

File: contrib/docker/Dockerfile [hunk 1]
```diff
@@ -1,13 +1,13 @@
-FROM python:3.6.3-alpine
+FROM python:3.6.7-alpine
```
Comment (yagebu) [inline contrib/docker/Dockerfile]: Why not use a more up-to-date version? And why fix it in the first place (that will probably just lead to similar security problems in the future)?

Comment (adamgibbins) [reply contrib/docker/Dockerfile]: To go higher would mean moving to 3.7, I'm not familiar enough with Python to make the call of doing that. My interest was rather the newer alpine, not python -- but the hub only had newer alpines if we upgraded python too.

Comment (yagebu) [reply contrib/docker/Dockerfile]: The build's not deterministic in any case - go with python:alpine

File: contrib/docker/Dockerfile [hunk 2]
```diff
@@ -19,10 +19,10 @@ RUN cd /root \
         && rm fava/docs/changelog.rst \
         && python3 -mpip install ./fava \
         && python3 -mpip uninstall --yes pip \
-        && find /usr/local/lib/python3.6/site-packages -name *.so -print0|xargs -0 strip -v \
+        && find /usr/local/lib/python3.6/site-packages -name '*.so' -print0|xargs -0 strip -v \
+        && find /usr/local/lib/python3.6/site-packages -name '*.pyc' -delete \
```
Comment (yagebu) [inline contrib/docker/Dockerfile]: This was intentionally removed - don't delete Python bytecode

**Comment by yagebu** [comment]
@adamgibbins: ping

File: contrib/docker/Dockerfile (hunk 2 above)
Comment (adamgibbins) [reply contrib/docker/Dockerfile]: Sorry I didn't notice it was previously removed. Can I ask why? The bytecode is regenerated on first execution, deleting upfront reduces the size of the image thats hanging around and duplicated multiple places by a fair bit at cost of slightly slower startup.

**Comment by yagebu** [comment]
I'll replace the Dockerfile with a simpler one to make it more easily maintainable

**Pull request closed without merging**